### ISSUE TITLE
Unable to archive application for ad-hoc distribution with Smartling framework

### ISSUE THREAD

**Issue by JaviSoto** [Smartling/ios-lqa-sdk#9]
We're hitting an error when archiving and exporting an app with the Smartling framework. The repro steps are:

- Download attached Xcode project (created with Xcode 9.2): [SmartlingArchiveTestApp.zip](https://github.com/Smartling/ios-lqa-sdk/files/2138559/SmartlingArchiveTestApp.zip)
- Change the bundle identifier and signing team so that it can be archived and signed.
- Select Product > Archive from the menu.
- When it finishes, select "export" and "ad-hoc distribution".

You will see an error like this:

<img width="770" alt="screen shot 2018-06-26 at 12 05 59 pm" src="https://user-images.githubusercontent.com/666807/41933956-31350574-793a-11e8-8d9e-5c5eca03ffb1.png">

The logs don't contain much more information:

```
2018-06-26 19:05:49 +0000  Validating IPA structure...
2018-06-26 19:05:50 +0000  error: Failed to verify bitcode in Smartling.framework/Smartling:
error: Bundle only contains bitcode-marker /var/folders/bl/nr03g5816dd2yxc_7cg4h111ffq00d/T/IDEDistributionOptionThinning.uiW/Payload/SmartlingArchiveTestApp.app/Frameworks/Smartling.framework/Smartling (arm64)

2018-06-26 19:05:50 +0000  /Applications/Xcode.app/Contents/Developer/usr/bin/ipatool exited with 1
2018-06-26 19:05:50 +0000  ipatool JSON: {
    alerts =     (
                {
            code = 2817;
            description = "Configuration issue: platform AppleTVSimulator.platform doesn't have any non-simulator SDKs; ignoring it";
            info =             {
            };
            level = WARN;
        },
                {
            code = 2817;
            description = "Configuration issue: platform iPhoneSimulator.platform doesn't have any non-simulator SDKs; ignoring it";
            info =             {
            };
            level = WARN;
        },
                {
            code = 2817;
            description = "Configuration issue: platform WatchSimulator.platform doesn't have any non-simulator SDKs; ignoring it";
            info =             {
            };
            level = WARN;
        },
                {
            code = 0;
            description = "Failed to verify bitcode in Smartling.framework/Smartling:\nerror: Bundle only contains bitcode-marker /var/folders/bl/nr03g5816dd2yxc_7cg4h111ffq00d/T/IDEDistributionOptionThinning.uiW/Payload/SmartlingArchiveTestApp.app/Frameworks/Smartling.framework/Smartling (arm64)\n\n";
            info =             {
            };
            level = ERROR;
            type = "malformed-payload";
        }
    );
}
```

How was the Smartling framework built and archived? Our theory is that it was built with `ENABLE_BITCODE=YES`, but the result of only compiling, not archiving, was posted online.

**Comment by JaviSoto** [comment]
This workaround worked: I ran this command to strip out bitcode from the smartling framework:

```bash
xcrun bitcode_strip -r Pods/Smartling-framework/Smartling.framework/Smartling -o Pods/Smartling-framework/Smartling.framework/Smartling
```

If Smartling wants to support Bitcode, I think it should upload a `.framework` that's the result of an archive, not just a regular build. Alternatively, it can just disable bitcode (`ENABLE_BITCODE=NO`)

**Comment by enegron-sl** [comment]
Thanks Javier, glad to hear you found a workaround.  If you have any further issues you might also try our standard support channel: support.smartling.com.

**Comment by foo4u** [comment]
As @enegron-sl mentioned, feel free to contact support about any future issues with the SDK. Since you've worked around this limitation, I'm closing this issue.

**Comment by JaviSoto** [comment]
Hi!

I did find a work-around, but the issue still remains in the library, and other people will run into it. Also, we would like to not have to keep the work-around seems it adds complexity to our build scripts.

I would ask that you guys keep this issue open until it's actually fixed. If other people run into this, this will be the first place they'll look, and at least they'll find my work-around in the open issue.

**Comment by foo4u** [comment]
@JaviSoto we believe this is fixed in 1.4.2.
